### ISSUE TITLE
Error when running kMP2 after KHF calc where GDF integrals were read in

### ISSUE THREAD

**Issue by verena-neufeld** [pyscf/pyscf#1163]
I am getting this error:

```
File "<dir>/pyscf/examples/pbc/35-gaussian_density_fit.py", line 28, in <module>
    my_mp.kernel()
  File "<dir>/pyscf/pyscf/pbc/mp/kmp2.py", line 757, in kernel
    kernel(self, mo_energy, mo_coeff, verbose=self.verbose, with_t2=with_t2)
  File "<dir>/pyscf/pyscf/pbc/mp/kmp2.py", line 72, in kernel
    naux = mp._scf.with_df.auxcell.nao_nr()
AttributeError: 'NoneType' object has no attribute 'nao_nr'
```

when running a kMP2 calculation after having run a KRHF calculation with GDF where the GDF integrals were read in.
This code which was partially copied and modified from `pbc/35-gaussian_density_fit.py` example (and possibly other examples) shows this:

```
from pyscf.pbc import df, gto, mp, scf

cell = gto.Cell()
cell.atom='''
C 0.000000000000   0.000000000000   0.000000000000
C 1.685068664391   1.685068664391   1.685068664391
'''
cell.basis = 'gth-szv'
cell.pseudo = 'gth-pade'
cell.a = '''
0.000000000, 3.370137329, 3.370137329
3.370137329, 0.000000000, 3.370137329
3.370137329, 3.370137329, 0.000000000'''
cell.unit = 'B'
cell.verbose = 5
cell.build()

mf = scf.KRHF(cell, cell.make_kpts([1,1,1])).density_fit()
mf.with_df._cderi_to_save = 'pbc_gdf.h5'
mf.kernel()

mf_2 = scf.KRHF(cell, cell.make_kpts([1,1,1])).density_fit()
mf_2.with_df._cderi = 'pbc_gdf.h5'
mf_2.kernel()

my_mp = mp.KMP2(mf_2)
my_mp.kernel()
```

**Comment by xwang862** [comment]
Hi Verena, thanks for reporting it. This occurs because GDF has not been built (since it already has precomputed `_cderi`) and thus its `auxcell` is None. 

Should be fixed by pullreq #1165.

**Comment by verena-neufeld** [comment]
Thanks for the quick reply and PR!
